psychopy/experiment/components/button/__init__.py

```diff
@@ -0,0 +1,210 @@
+            hint=_translate("Should text anchor to the top, center or bottom of the box?"),
+            label=_localized['anchor'])
+        self.params['borderWidth'] = Param(
+            borderWidth, valType='num', inputType="single", allowedTypes=[], categ='Appearance',
+            updates='constant', allowedUpdates=_allow3[:],
+            hint=_translate("Textbox border width"),
+            label=_localized['borderWidth'])
+
+
+    def writeInitCode(self, buff):
+        # do we need units code?
+        if self.params['units'].val == 'from exp settings':
+            unitsStr = ""
+        else:
+            unitsStr = "units=%(units)s," % self.params
+        # do writing of init
+        inits = getInitVals(self.params, 'PsychoPy')
+        code = (
+                       "%(name)s = visual.ButtonStim(win, \n"
+                       "    text=%(text)s, font=%(font)s,\n"
+                       "     pos=%(pos)s," + unitsStr + "\n"
+                       "     letterHeight=%(letterHeight)s,\n"
+                       "     size=%(size)s, borderWidth=%(borderWidth)s,\n"
+                       "     fillColor=%(fillColor)s, borderColor=%(borderColor)s,\n"
+                       "     color=%(color)s, colorSpace=%(colorSpace)s,\n"
+                       "     opacity=%(opacity)s,\n"
+                       "     bold=%(bold)s, italic=%(italic)s,\n"
+                       "     padding=%(padding)s,\n"
+                       "     anchor=%(anchor)s,\n"
+                       "     name='%(name)s')"
+        )
+        buff.writeIndentedLines(code % inits)
+
+    def writeFrameCode(self, buff):
+        BaseVisualComponent.writeFrameCode(self, buff)
+        # do writing of init
+        inits = getInitVals(self.params, 'PsychoPy')
+        callback = str(inits['callback'])
+        if inits['callback']:
+            # Add indent to any linebreaks to account for the loop
+            callback = str(inits['callback']).replace("\n", "\n   ")
+        else:
+            callback = "pass"
+        if inits['forceEndRoutine'].val:
+            endRt = "continueRoutine = False"
+        else:
+            endRt = ""
+        if inits['oncePerClick'].val:
+            hammingStr = "   while %(name)s.isClicked:\n      pass\n"
+        else:
+            hammingStr = ""
+        code = (
+            "# check whether button \"%(name)s\" has been pressed\n"
+            "if %(name)s.isClicked:\n" + hammingStr +
+            "   " + callback + "\n" +
+            "   " + endRt + "\n"
+        )
+        buff.writeIndentedLines(code % inits)
+
+    def writeRoutineEndCode(self, buff):
+        BaseVisualComponent.writeRoutineEndCode(self, buff)
+        if len(self.exp.flow._loopList):
+            currLoop = self.exp.flow._loopList[-1]  # last (outer-most) loop
+        else:
+            currLoop = self.exp._expHandler
+        name = self.params['name']
+        code = f"{currLoop.params['name']}.addData('{name}.rt', t)\n"
```

Comment:
This looks like you're adding just the time of the end of the routine rather then the time the button was pressed. Maybe check the time code for mouse rt storage